test/jdk/java/foreign/TestNulls.java

```diff
@@ -0,0 +1,223 @@
+import org.testng.annotations.NoInjection;
+import org.testng.annotations.Test;
+
+import java.lang.constant.Constable;
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.MethodHandles;
+import java.lang.invoke.VarHandle;
+import java.lang.ref.Cleaner;
+import java.lang.reflect.Array;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.nio.channels.FileChannel;
+import java.nio.file.Path;
+import java.util.*;
+import java.util.function.UnaryOperator;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static org.testng.Assert.*;
+import static org.testng.Assert.fail;
+
+/**
+ * This test makes sure that public API classes (listed in {@link TestNulls#CLASSES}) throws NPEs whenever
+ * nulls are provided. The test looks at all the public methods in all the listed classes, and injects
+ * values automatically. If an API takes a reference, the test will try to inject nulls. For APIs taking
+ * either reference arrays, or collections, the framework will also generate additional <em>replacements</em>
+ * (e.g. other than just replacing the array, or collection with null), such as an array or collection
+ * with null elements. The test can be customized by adding/removing classes to the {@link #CLASSES} array,
+ * by adding/removing default mappings for standard carrier types (see {@link #DEFAULT_VALUES} or by
+ * adding/removing custom replacements (see {@link #REPLACEMENT_VALUES}).
+ */
+public class TestNulls {
+
+    static final Class<?>[] CLASSES = new Class<?>[] {
+            MemorySegment.class,
+            MemoryAddress.class,
+            MemoryLayout.class,
+            MemoryLayout.PathElement.class,
+            SequenceLayout.class,
+            ValueLayout.class,
+            GroupLayout.class,
+            Addressable.class,
+            MemoryAccess.class,
+            MappedMemorySegments.class,
+            MemoryLayouts.class,
+            MemoryHandles.class
+    };
+
+    static final Set<String> EXCLUDE_LIST = Set.of(
+            "jdk.incubator.foreign.MemoryLayout/withAttribute(java.lang.String,java.lang.constant.Constable)/1/0",
+            "jdk.incubator.foreign.MemoryAddress/asSegmentRestricted(long,java.lang.Runnable,java.lang.Object)/1/0",
+            "jdk.incubator.foreign.MemoryAddress/asSegmentRestricted(long,java.lang.Runnable,java.lang.Object)/2/0",
+            "jdk.incubator.foreign.SequenceLayout/withAttribute(java.lang.String,java.lang.constant.Constable)/1/0",
+            "jdk.incubator.foreign.ValueLayout/withAttribute(java.lang.String,java.lang.constant.Constable)/1/0",
+            "jdk.incubator.foreign.GroupLayout/withAttribute(java.lang.String,java.lang.constant.Constable)/1/0",
+            "jdk.incubator.foreign.MemoryHandles/insertCoordinates(java.lang.invoke.VarHandle,int,java.lang.Object[])/2/1"
+    );
```

Comment:
This is the set of API points, and parameter positions, that accept null. Cool to have this in one place.

Reply:
Yeah - it's a bit "stringy" and maybe can be improved, but compared to what we had before, I think it's a massive improvement :-)

If you are curious about the encoding it's:

<className> `/` <method signature> `/` <index of the parameter to be replaced> `/` <index of the replacement to be used>

Reply:
Could you add this in a comment as well?